apps/emqx/src/emqx_persistent_session_ds.erl

```diff
@@ -130,6 +139,62 @@ do_open_iterator(TopicFilter, StartMS, IteratorID) ->
     {ok, _It} = emqx_ds_storage_layer:ensure_iterator(?DS_SHARD, IteratorID, Replay),
     ok.
 
+-spec del_subscription(emqx_ds:iterator_id() | undefined, emqx_types:topic(), emqx_ds:session_id()) ->
+    ok | {skipped, disabled}.
+del_subscription(IteratorID, TopicFilterBin, DSSessionID) ->
+    ?WHEN_ENABLED(
+        begin
+            TopicFilter = emqx_topic:words(TopicFilterBin),
+            Ctx = #{iterator_id => IteratorID},
+            case IteratorID of
+                undefined ->
```

Comment:
Nit. Looks unusual to handle `undefined` on that level. I'd move that to `emqx_session` level.

Reply:
This was mostly to have idempotency for calls to this function: if this happens to be called with a session with this particular iterator id already removed (so, nothing to close), we still want to be sure to remove the iterator reference (that depends only the topic filter) from `?ITERATOR_REF_TAB`.